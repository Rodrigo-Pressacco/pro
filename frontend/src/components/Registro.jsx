import React, { useState, useEffect } from "react";
import {useForm} from "react-hook-form"
import axios from 'axios';
import "../styles/Registro.css";
import {Link} from 'react-router-dom'
import url from "../utils/datosInicio.json"


const Registro = (props) => {

  const confirmacionEmail = {
    subject: "Confirmacion de Registro NeoTravel",
    body: "Te has registrado correctamente en nuestro servicio. Gracias por unirte a NeoTravel, "
  }

  // const [registroBoton,setRegistroBoton] = useState(true)
  const {register,formState:{errors},getValues, handleSubmit}= useForm( {mode: "onChange"});
  const onSubmit = (data,e) =>{
    e.preventDefault();
    console.log(data)
    return data
  }

  const [datosUsuarioTraidos, setdatosUsuarioTraidos] = useState([{
    id: null,
    email: null
}]);

useEffect(() => {
    if (datosUsuarioTraidos[0].email === null) {
        axios.get(`http://${url.hostUrl}:8080/usuarios/todos`)
            .then((response) => {
                setdatosUsuarioTraidos(response.data)
            })
    }
});

  const [formOk,setFormOk] = useState(true)
  const checkPassword = (password) =>{ 
    return password.length >6;
  }
  const checkNombre = (password) =>{ 
    return password.length >2;
  }

  const checkEmail =(email) =>{
    let arra=[]
    datosUsuarioTraidos.filter((e)=>{
      if (e.email===email) {
        arra.push(e)
      }
    })
    const regex = /\w+@[A-z]+.[A-z]{2,3}(.[a-z]{2,3})?/
    if(arra.length===0 && regex.test(email)){
      
      return true
    }else{
      return false;
    }   
    
  }
  const checkApellido = (apellido) =>{ 
    return apellido.length >2;
  }
  
  const checkConfirmPass = (password,confirm) => {
    return password === confirm && confirm.length > 6
  }
  

  const validarFormulario = () => {
    let status = true

    status *= checkNombre(document.getElementById('registro-nombre').value)
    status *= checkApellido(document.getElementById('registro-apellido').value)
    status *= checkEmail(document.getElementById('registro-email').value)
    status *= checkPassword(document.getElementById('registro-password').value)
    status *= checkConfirmPass(document.getElementById('registro-password').value,document.getElementById('registro-confirmPass').value)
    
    if (status === 1){
      setFormOk(false)
    } else {
      setFormOk(true)
    }

  }

  const handleError = (id) => {
    let el = document.getElementById(id);
    if(el.value.length ===0){
    el.classList.add("registro-error")
    }
    else{
      el.classList.remove("registro-error")
    }
  }
  const handleClean = (id) => {
    let el = document.getElementById(id);
     el.classList.remove("registro-error")
  }
  const existingErrors =()=>{
    return Object.keys(errors).length >0
  }

  const cambiarBotenes = (datos)=>{
    props.cambiarBotton(datos)
    //console.log(datos);

    axios.post(`http://${url.hostUrl}:8080/usuarios/nuevo`, {
      nombre: document.getElementById('registro-nombre').value,
      apellido: document.getElementById('registro-apellido').value,
      email: document.getElementById('registro-email').value,
      password: document.getElementById('registro-password').value
  })
    .then((response) => {
      axios.post(`http://${url.hostUrl}:8080/email/verificacion/${response.data.email}/${confirmacionEmail.subject}/${confirmacionEmail.body}`)
      console.log(response)
  })
} 

  return (
    <div className="div-registro">
      <h1>Crear Cuenta</h1>
      {console.log()}
      <form action="" method="POST" className="form-registro" onSubmit={handleSubmit(onSubmit)} onMouseOver={validarFormulario}>
        <div className="divGeneral-nombre-apellido">
          <div className="divParticular-nombre-apellido">
            <label >Nombre</label>
            <input type="text" className="input-nombre-apellido" {...register("registronombre", {required: "Campo obligatorio"})} id="registro-nombre" onBlur={()=>handleError("registro-nombre")} onFocus={()=>handleClean("registro-nombre")}/>
            {errors.registronombre && <span className="errorsForm">{errors.registronombre.message}</span>}
          </div>
          <div className="divParticular-nombre-apellido">
            <label >Apellido</label>
            <input type="text" className="input-nombre-apellido" {...register("registroapellido", {required: "Campo obligatorio"})} id="registro-apellido" onBlur={()=>handleError("registro-apellido")} onFocus={()=>handleClean("registro-apellido")}/>
            {errors.registroapellido && <span className="errorsForm">{errors.registroapellido.message}</span>}
          </div>
        </div>
        <div className="div-email-password">
          <div className="divParticular-email-password">
            <label >Email</label>
            <input type="email" className="input-email-password" {...register("registroemail", 
            { pattern: { value:/\w+@[A-z]+.[A-z]{2,3}(.[a-z]{2,3})?/,
                         message:"Email no válido"},
              required:"Campo obligatorio"
              })} id="registro-email" onBlur={()=>handleError("registro-email")} onFocus={()=>handleClean("registro-email")}/>
          {errors.registroemail && <span className="errorsForm">{errors.registroemail.message}</span>}
          </div>
          <div className="divParticular-email-password">
            <label >Contraseña</label>
            <input type="password" className="input-email-password" {...register("registropassword",
            {required:"Campo Obligatorio",
            minLength:{
                value:6,
                message:"La contraseña debe contener al menos 6 caracteres"}})} id="registro-password" onBlur={()=>handleError("registro-password")} onFocus={()=>handleClean("registro-password")}/>
            {errors.registropassword && <span className="errorsForm">{errors.registropassword.message}</span>}
          </div>
          <div className="divParticular-email-password">
            <label >Repetir Contraseña</label>
            <input type="password" className="input-email-password" {...register("registrorepetirPassword",{required:"Confirmar contraseña", validate:{
              checkPasswordsMatch :(repeatedPass)=>{
                const {registropassword} = getValues()
                return registropassword === repeatedPass || "Las contraseñas no coinciden"
              }}})} id="registro-confirmPass" onBlur={()=>handleError("registro-confirmPass")} onFocus={()=>handleClean("registro-confirmPass")}/>
            {errors.registrorepetirPassword && <span className="errorsForm">{errors.registrorepetirPassword.message}</span>}
          </div>
        </div>
        <div className="div-button">
          {formOk ? <button disabled="disabled">Crear Cuenta</button>: <Link to="/login" onClick={()=>cambiarBotenes(3)} className="register-button">Crear Cuenta</Link>}
          <p>¿Ya tenienes una cuenta? <Link to="/login" onClick={()=>cambiarBotenes(3)} >Iniciar Sesion</Link> </p>
        </div>
      </form>
    </div>

  )
}

export default Registro;