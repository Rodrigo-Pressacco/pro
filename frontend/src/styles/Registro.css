.div-registro{
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 78px;
}

.div-registro h1{
    color: var(--main-color);
    font-style: normal;
    font-weight: bold;
    font-size: 24px;
    line-height: 28px;
    text-align: center;
}

.form-registro{
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
}

.divGeneral-nombre-apellido{
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 448px;
}

.div-email-password{
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 448px;
}

.div-button{
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 448px;
    padding-top: 29px;
    margin-bottom: 77px;
    
    color: var(--secondary-color);
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
    text-align: center;
}

.divParticular-nombre-apellido{
    display: flex;
    flex-direction: column;
    width: 46.5%;
}

.divParticular-email-password{
    width: 100%;
}

.input-nombre-apellido,
.input-email-password{
    height: 40px;
    padding-left: 13px;
    background: #FFFFFF;
    box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
    border-radius: 5px;
    border: none;
}

.input-nombre-apellido,
.input-email-password{
    width: 100%;
    margin: 15px 0;
}

.input-nombre-apellido,
.input-email-password,
.form-registro label{
    color: var(--secondary-color);
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
}

.form-registro label{
    margin-top: 17px;
}

.div-button button,.register-button{
    background: var(--main-color);
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.12);
    border-radius: 5px;
    border: none;
    height: 40px;
    width: 46%;
    margin-bottom: 15px;
    
    font-style: normal;
    font-weight: bold;
    font-size: 16px;
    line-height: 19px;
    text-align: center;
    color: #FFFFFF;
}
.register-button{
    text-decoration: none;
    line-height: 40px;
}
.errorsForm{
    display: block;
    text-align: end;
    color: #FF0000;
    font-size: 12px;
}
.registro-error{    
    background: #FFE1E1;
    border: 1px solid #FF0000;
    box-sizing: border-box;
    box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
}
@media (max-width: 777px){
    .divGeneral-nombre-apellido{
        flex-direction: column;
        width: 346px;
    }
    
    .divParticular-nombre-apellido,
    .div-button button {
        width: 100%;
    }
    
    .div-email-password,
    .div-button {
        width: 346px;
    }
}

