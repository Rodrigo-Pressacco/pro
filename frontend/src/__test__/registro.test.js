import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Registro from '../components/Registro'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<Registro />', () => {
    let registro
    let onSubmit = jest.fn() 
    let checkPassword = jest.fn() 
    let checkNombre = jest.fn() 
    let checkEmail = jest.fn()
    let checkApellido = jest.fn() 
    let checkConfirmPass = jest.fn()
    let validarFormulario = jest.fn() 
    let handleError = jest.fn() 
    let handleClean = jest.fn()   
    let existingErrors = jest.fn()   
    let cambiarBotones = jest.fn(); 
    
    beforeEach(() => {
        registro = shallow(<Registro    onSubmit={onSubmit}
                                        checkPassword={checkPassword}
                                        checkNombre={checkNombre}
                                        checkEmail={checkEmail}
                                        checkApellido={checkApellido}
                                        checkConfirmPass={checkConfirmPass}
                                        validarFormulario={validarFormulario}
                                        handleError={handleError}
                                        handleClean={handleClean}
                                        existingErrors={existingErrors}
                                        cambiarBotones={cambiarBotones} />)
    });

    it("Deberia mostrar <Registro/> correctamente", () => {
        expect(registro).toMatchSnapshot();
    });

})