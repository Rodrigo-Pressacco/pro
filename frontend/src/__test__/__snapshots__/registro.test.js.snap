// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Registro /> Deberia mostrar <Registro/> correctamente 1`] = `
<div
  className="div-registro"
>
  <h1>
    Crear Cuenta
  </h1>
  <form
    action=""
    className="form-registro"
    method="POST"
    onMouseOver={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="divGeneral-nombre-apellido"
    >
      <div
        className="divParticular-nombre-apellido"
      >
        <label>
          Nombre
        </label>
        <input
          className="input-nombre-apellido"
          id="registro-nombre"
          name="registronombre"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
        />
      </div>
      <div
        className="divParticular-nombre-apellido"
      >
        <label>
          Apellido
        </label>
        <input
          className="input-nombre-apellido"
          id="registro-apellido"
          name="registroapellido"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
        />
      </div>
    </div>
    <div
      className="div-email-password"
    >
      <div
        className="divParticular-email-password"
      >
        <label>
          Email
        </label>
        <input
          className="input-email-password"
          id="registro-email"
          name="registroemail"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="email"
        />
      </div>
      <div
        className="divParticular-email-password"
      >
        <label>
          Contraseña
        </label>
        <input
          className="input-email-password"
          id="registro-password"
          name="registropassword"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="password"
        />
      </div>
      <div
        className="divParticular-email-password"
      >
        <label>
          Repetir Contraseña
        </label>
        <input
          className="input-email-password"
          id="registro-confirmPass"
          name="registrorepetirPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="password"
        />
      </div>
    </div>
    <div
      className="div-button"
    >
      <button
        disabled="disabled"
      >
        Crear Cuenta
      </button>
      <p>
        ¿Ya tenienes una cuenta? 
        <Link
          onClick={[Function]}
          to="/login"
        >
          Iniciar Sesion
        </Link>
         
      </p>
    </div>
  </form>
</div>
`;
